dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift

```diff
@@ -115,55 +132,70 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
-            viewModel?.takeProfitStopLossInputAreaViewModel?.takeProfitAlert = nil
-        }
-        if let slUsdcDiff = triggerOrdersInput?.stopLossOrder?.price?.usdcDiff?.doubleValue, slUsdcDiff < 0 {
-            viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossAlert = .init(.init(title: "test error bad stop loss",
-                                                                                           body: "test error stop loss description",
-                                                                                           level: .error))
-        } else {
-            viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossAlert = nil
-        }
-        #endif
 
+        // update displayed values
         viewModel?.takeProfitStopLossInputAreaViewModel?.takeProfitPriceInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.takeProfitOrder?.price?.triggerPrice?.doubleValue, digits: 2)
         viewModel?.takeProfitStopLossInputAreaViewModel?.gainInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.takeProfitOrder?.price?.usdcDiff?.doubleValue, digits: 2)
         viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossPriceInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.stopLossOrder?.price?.triggerPrice?.doubleValue, digits: 2)
         viewModel?.takeProfitStopLossInputAreaViewModel?.lossInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.stopLossOrder?.price?.usdcDiff?.doubleValue, digits: 2)
 
-        // TODO: add to this for stop limit and take prof limit
-        if let takeProfitOrderTriggerPrice = triggerOrdersInput?.takeProfitOrder?.price?.triggerPrice?.doubleValue, !takeProfitOrderTriggerPrice.isZero {
+        // update order types
+        if let _ = triggerOrdersInput?.takeProfitOrder?.price?.limitPrice?.doubleValue {
+            AbacusStateManager.shared.triggerOrders(input: Abacus.OrderType.takeprofitlimit.rawValue, type: .takeprofitordertype)
+        } else {
             AbacusStateManager.shared.triggerOrders(input: Abacus.OrderType.takeprofitmarket.rawValue, type: .takeprofitordertype)
         }
-        if let stopLossOrderTriggerPrice = triggerOrdersInput?.stopLossOrder?.price?.triggerPrice?.doubleValue, !stopLossOrderTriggerPrice.isZero {
+        if let _ = triggerOrdersInput?.stopLossOrder?.price?.limitPrice?.doubleValue {
+            AbacusStateManager.shared.triggerOrders(input: Abacus.OrderType.stoplimit.rawValue, type: .stoplossordertype)
+        } else {
             AbacusStateManager.shared.triggerOrders(input: Abacus.OrderType.stopmarket.rawValue, type: .stoplossordertype)
         }
+
+        if let error = errors.first {
+            viewModel?.submissionReadiness = .fixErrors(cta: error.resources.action?.localized)
+        } else if triggerOrdersInput?.takeProfitOrder?.price?.triggerPrice?.doubleValue == nil
+            && triggerOrdersInput?.takeProfitOrder?.orderId == nil
```

Comment:
I am not doing this check on Android.  It seems to ready for submission if there is no ValidationError.

Reply:
went over this with mulan today, without this check, the "Add triggers" button is enabled even with empty input. I don't think that is great UX, what do you think?

We discussed adding something to abacus like `containsChanges` to denote whether that button should be enabled or not, even with no errors